pkg/sql/compile/compile.go

```diff
@@ -505,7 +505,7 @@ func (c *Compile) compilePlanScope(n *plan.Node, ns []*plan.Node) ([]*Scope, err
 }
 
 func (c *Compile) compileExternScan(n *plan.Node) []*Scope {
-	ds := &Scope{Magic: Normal}
+	ds := &Scope{Magic: ExternalScan}
```

Comment:
I think you should deal with parallel logic here and still use the Normal but not add a new ExternalRun.
just refer to compileTableScan.

Reply:
ok